File: src/changecolorsrv.py
#!/usr/bin/env python3

#chmod u+x ~/catkin_ws/src/beginner_tutorials/src/changecolorsrv.py

import rospy
from std_srvs.srv import Empty
from turtlesim.srv import*


rospy.init_node('setcolor' , anonymous = True)


color_r = rospy.get_param("r")
color_g = rospy.get_param("g")
color_b = rospy.get_param("b")

print(color_r,color_g,color_b)

rospy.wait_for_service("clear")

print("clear working")

rospy.set_param("background_r" , color_r)
rospy.set_param("background_g" , color_g)
rospy.set_param("background_b" , color_b)

clear_background=rospy.ServiceProxy("clear", Empty)

clear_background()

File: src/changecolorsrv.cpp
// Umut Emre Celik 18070006039

#include <ros/ros.h>
#include <std_srvs/Empty.h>
#include <turtlesim/Spawn.h>

int main(int argc, char** argv)
{
	ros::init(argc,argv,"setcolor");
	ros::NodeHandle n;
	int color_r, color_g ,color_b;
	n.getParam("r" , color_r);
	n.getParam("g" , color_g);
	n.getParam("b" , color_b);

	ros::service::waitForService("clear");
	ros::param::set("background_r", color_r);
	ros::param::set("background_g", color_g);
	ros::param::set("background_b", color_b);

	ros::ServiceClient clearclient = n.serviceClient<std_srvs::Empty>("clear");
	std_srvs::Empty srv;

	clearclient.call(srv);

	return 0;
}